source/localizable/tutorial/service.md

```diff
@@ -0,0 +1,293 @@
+    this.assert.ok(location, 'createMap called with location');
+    return DUMMY_ELEMENT;
+  }
+});
+
+moduleFor('service:maps', 'Unit | Service | maps', {
+  needs: ['util:google-maps']
+});
+
+test('should create a new map if one isnt cached for location', function (assert) {
+  assert.expect(4);
+  let stubMapUtil = MapUtilStub.create({ assert });
+  let mapService = this.subject({ mapUtil: stubMapUtil });
+  let element = mapService.getMapElement('San Francisco');
+  assert.ok(element, 'element exists');
+  assert.equal(element.className, 'map', 'element has class name of map');
+});
+
+test('should use existing map if one is cached for location', function (assert) {
+  assert.expect(1);
+  let stubCachedMaps = Ember.Object.create({
+    sanFrancisco: DUMMY_ELEMENT
+  });
+  let mapService = this.subject({ cachedMaps: stubCachedMaps });
+  let element = mapService.getMapElement('San Francisco');
+  assert.equal(element, DUMMY_ELEMENT, 'element fetched from cache');
+});
+```
+
+Note that the test uses a dummy object as the returned map element.  This can be any object because its only used to assert that the cache has been accessed.
+Also note that the location has been `camelized` in the cache object, so that it may be used as a key.
+
+Now implement the service as follows.  Note that we check if a map already exists for the given location and use that one, otherwise we call a Google Maps utility to create one.
+We abstract our interaction with the maps API behind an Ember utility so that we can test our service without making network requests to Google.
+
+```app/services/maps.js
+import Ember from 'ember';
+import MapUtil from '../utils/google-maps';
+
+export default Ember.Service.extend({
+
+  init() {
+    if (!this.get('cachedMaps')) {
+      this.set('cachedMaps', Ember.Object.create());
+    }
+    if (!this.get('mapUtil')) {
+      this.set('mapUtil', MapUtil.create());
+    }
+  },
+
+  getMapElement(location) {
+    let camelizedLocation = location.camelize();
+    let element = this.get(`cachedMaps.${camelizedLocation}`);
+    if (!element) {
+      element = this.createMapElement();
+      this.get('mapUtil').createMap(element, location);
+      this.set(`cachedMaps.${camelizedLocation}`, element);
+    }
+    return element;
+  },
+
+  createMapElement() {
+    let element = document.createElement('div');
+    element.className = 'map';
+    element.setAttribute('style', 'width:300px;height:300px;');
+    return element;
+  }
+
+});
+```
+
+### Making Google Maps Available
+
+Before implementing the map utility we need to make the 3rd party map API available to our Ember app.
+There are several ways to include 3rd party libraries in Ember. See the guides section on [managing dependencies](../../addons-and-dependencies/managing-dependencies/) as a starting point when you need to add one.
+
+Since Google provides its map API as a remote script, we'll use curl to download it into our project's vendor directory.
```

Comment:
I'm pretty sure its stable by version.  They up the version in the url when they make updates.  For the case of the tutorial I think its fine, since I or the typical tutorial user is not going to acquire a key from them just to do the tutorial sample.  Its worth considering the alternative though, which is linking to it in the index.html.  That way it would be easer to update an existing app (by just changing the url)